backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/entities/Shop.java

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.google.gpay.virtualqueue.backendservice.entities;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+public class Shop {
+    private String shopOwnerUid;
+    private String shopOwnerName;
+    private String shopName;
+    private String address;
+    private String phoneNumber;
+    private Integer lastTokenAlloted;
+    private String shopType;
+    private Integer shopId;
+    private List<UUID> tokens = new ArrayList<>();
+
+    public Shop(String shopOwnerUid, String shopOwnerName, String shopName, String address, String phnNo,
+            String shopType) {
+        super();
+        this.shopOwnerUid = shopOwnerUid;
+        this.shopOwnerName = shopOwnerName;
+        this.shopName = shopName;
+        this.address = address;
+        this.phoneNumber = phnNo;
+        this.shopType = shopType;
+    }
```

Comment:
Shall we replace this with @AllArgsConstructor ?

Reply:
In @AllArgsConstructor it passes all the variables that we initialized in the file in the constructor as well. But here some of those may not be initialized at the time we make the Shop entity like list of tokens and last_alloted_token. That's why thought of not using AllArgsContructor here.